prow/pjutil/pjutil.go

```diff
@@ -37,14 +37,41 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/downwardapi"
 )
 
+// Options allows a client to set some fields
+// when a ProwJob is being created.
+type Options struct {
+	state prowapi.ProwJobState
+}
+
+// Option configures an Options value
+type Option func(*Options)
```

Comment:
I guess would be good to invent some other name, more descriptive and less general.

Reply:
The purpose of this function is about modifying a ProwJob, so it can probably be better renamed with one of the following:
```golang
type Mutations struct { ... }
type Mutator func(*Mutations)
```
```golang
type Modifiers struct { ... }
type Modifier func(*Modifiers)
```
```golang
type Transformations struct { ... }
type Transform func(*Transformations)
```
```golang
type Appliers struct { ... }
type Apply func(*Appliers)
```
Note also I'm omitting any reference to `ProwJob` on purpose here, as `ProwJobOptions`. That's because the package name is `k8s.io/test-infra/prow/pjutil`, therefore the context is already in the calling site:
```golang
// One knows that Mutator has something to do with a ProwJob once he reads `pjutil`
var mutator pjutil.Mutator
```